### PR TITLE
[DOC] Adjust coverage for partitionBy()

### PR DESCRIPTION
**Pull request by ted-yu**
This is the related thread: http://search-hadoop.com/m/q3RTtO3ReeJ1iF02&subj=Re+partitioning+json+data+in+spark

Michael suggested fixing the doc.

Please review.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 1]
```diff
@@ -119,7 +119,7 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    * Partitions the output by the given columns on the file system. If specified, the output is
    * laid out on the file system similar to Hive's partitioning scheme.
    *
-   * This is only applicable for Parquet at the moment.
+   * This was initally applicable for Parquet but in 1.5.x covers JSON as well.
```
Comment (JoshRosen) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: 1.5.x or 1.5+?

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 2]
```diff
@@ -119,7 +119,7 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    * Partitions the output by the given columns on the file system. If specified, the output is
    * laid out on the file system similar to Hive's partitioning scheme.
    *
-   * This is only applicable for Parquet at the moment.
+   * This was initally applicable for Parquet but in 1.5+ covers JSON as well.
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: also "text"

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 3]
```diff
@@ -119,7 +119,7 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    * Partitions the output by the given columns on the file system. If specified, the output is
    * laid out on the file system similar to Hive's partitioning scheme.
    *
-   * This is only applicable for Parquet at the moment.
+   * This was initially applicable for Parquet but in 1.5+ covers JSON as well.
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: And text, ORC, and avro

**Comment by tedyu** [comment]
The test failure was not related to the patch.
Looks like HiveThriftBinaryServerSuite timed out:
```
[info] HiveThriftBinaryServerSuite:
[info] - GetInfo Thrift API (431 milliseconds)
[info] - JDBC query execution (9 seconds, 740 milliseconds)
[info] - Checks Hive version (1 second, 386 milliseconds)
[info] - SPARK-3004 regression: result set containing NULL (1 second, 685 milliseconds)
[info] - SPARK-4292 regression: result set iterator issue (5 seconds, 77 milliseconds)
[info] - SPARK-4309 regression: Date type support (1 second, 328 milliseconds)
[info] - SPARK-4407 regression: Complex type support (2 seconds, 165 milliseconds)
[info] - test multiple session (5 seconds, 857 milliseconds)
Attempting to post to Github...
 > Post successful.
Build step 'Execute shell' marked build as failure
```

**Comment by tedyu** [comment]
@marmbrus :
Is there anything I need to do ?

**Comment by tedyu** [comment]
@marmbrus
Gentle ping

**Comment by marmbrus** [comment]
Thanks, merged to master and 1.6

**Pull request closed without merging**